packages/runtime/container-runtime/src/containerRuntime.ts

```diff
@@ -2185,6 +2224,16 @@ export class ContainerRuntime
 	}
 
 	public setConnectionState(connected: boolean, clientId?: string) {
+		if (this.idCompressorMode === "delayed" && !this.delayedCompressorLoading) {
```

Comment:
This is the place I'm delay-loading ID compressor. From implementation detail, we should probably delay it up to a point where this client attempts to send first op. But op sending is synchronous, and it would be great to reserve IDs on very first op. Also, maybe in the future ID compressor will implement a schema where another "write" client could reserve IDs for "read" client - then we start using compressed IDs on any connection.